src/docs/generate_oci_documentation.py

```diff
@@ -0,0 +1,358 @@
+        """
+        Permit to get the arches associated to the tag
+        """
+        logging.info("Getting the arches for %s", tag)
+        arches = []
+        for arch in arches_list:
+            arches.append(arch["platform"]["architecture"])
+        return arches
+
+    def get_lowest_risk(self, tags: List[str]) -> str:
+        """
+        Get the lowest risk associated with the tag
+        """
+        risk_sorted = ["stable", "candidate", "beta", "edge"]
+
+        all_tags_str = " ".join(tags)
+        for risk in risk_sorted:
+            if risk in all_tags_str:
+                return risk
+
+        return "edge"
+
+    def get_tags(self, tag: str, all_tags: List[str]) -> List[str]:
+        """
+        Permit to get the tags associated to the track-base
+        """
+        return [_tag for _tag in all_tags if tag.split("_")[0] == _tag.split("_")[0]]
+
+    def build_releases_data(
+        self,
+        release_lts: Dict,
+        all_tags: List[str],
+    ) -> List[Dict]:
+        """Build the releases info data structure"""
+        releases = []
+        tracks_base = []
+        for tag in all_tags:
+            release_data = {}
+            track = tag.split("-")[0]
+            base = tag.split("_")[0].split("-")[1]
+
+            if tag.split("_")[0] in tracks_base:
+                continue
+
+            tracks_base.append(tag.split("_")[0])
+            release_data["track"] = track
+            release_data["base"] = base
+
+            if release_lts[base]:
+                release_data["type"] = "LTS"
+
+            manifest_list = self.run_skopeo_command(
+                "inspect", [f"docker://{self.url}:" + tag, "--raw"]
+            )
+
+            if "manifests" in manifest_list:
+                release_data["architectures"] = self.get_arches(
+                    tag, manifest_list["manifests"]
+                )
+            else:
+                manifest_list = self.run_skopeo_command(
+                    "inspect", [f"docker://{self.url}:" + tag]
+                )
+
+                release_data["architectures"] = [manifest_list["Architecture"]]
+
+            release_data["tags"] = self.get_tags(tag, all_tags)
+
+            release_data["risk"] = self.get_lowest_risk(release_data["tags"])
+            releases.append(release_data)
+
+        return releases
+
+    def read_data_template(self) -> Dict:
+        """Reads and parses the YAML contents of the data template"""
+        template_file = f"{self.data_dir}/{self.name_of_oci}/documentation.yaml"
+        logging.info("Opening the template file %s", template_file)
+        with open(template_file, "r", encoding="UTF-8") as file:
+            try:
+                template_data = yaml.safe_load(file) or {}
+            except yaml.YAMLError as exc:
+                logging.error("Error when loading the ubuntu template file")
+                raise exc
+            try:
```

Comment:
you are loading the data twice. if you're going to instantiate the class object, then you'll automatically have the JSON out of it and you don't need the `yaml.safe_load` above. Just do `.dict(exclude_none=True)`

Reply:
Could you provide an example to help me better understand?

Reply:
Here: https://github.com/canonical/rockcraft/blob/4c458b4f55efbcbf1403056c13d6dfefcc2eb8af/rockcraft/lifecycle.py#L175

Reply:
Thank you for sharing the link. Can you kindly assist me in identifying the location of the code where the data is loaded twice?

Reply:
Done